chains/manager.go

```diff
@@ -591,7 +586,7 @@ func (m *manager) AddRegistrant(r Registrant) {
 func (m *manager) createAvalancheChain(
 	ctx *snow.ConsensusContext,
 	genesisData []byte,
-	vdrs validators.Set,
+	vdrs validators.Manager,
```

Comment:
I will respond for all overriden comments here:

I don't think we should use the overridden manager in normal paths. If something goes off, we would never be able to debug it. We could have directly call manager with explicit primary network ID in those components (which I did before this [commit](https://github.com/ava-labs/avalanchego/pull/1857/commits/bc96bf1405f960f7e8a241ab70d2ff7b5dc90875)), but it would be too specific. I feel like we should keep calling `constants.PrimaryNetworkID` as minimal and as a param for components. So once we feel like we should be using those components for subnets, we could easily use same components.